power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -289,50 +280,63 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
     // helper function to update vectors of components
     template <class CompType, cache_type_c CacheType>
     void update_component(std::vector<typename CompType::UpdateType> const& components,
-                          std::vector<Idx2D> const& sequence_idx) {
+                          std::span<Idx2D const> sequence_idx) {
         if (!components.empty()) {
             update_component<CompType, CacheType>(components.begin(), components.end(), sequence_idx);
         }
     }
     template <class CompType, cache_type_c CacheType>
     void update_component(std::span<typename CompType::UpdateType const> components,
-                          std::vector<Idx2D> const& sequence_idx) {
+                          std::span<Idx2D const> sequence_idx) {
         if (!components.empty()) {
             update_component<CompType, CacheType>(components.begin(), components.end(), sequence_idx);
         }
     }
     template <class CompType, cache_type_c CacheType>
     void update_component(ConstDataset::RangeObject<typename CompType::UpdateType const> components,
-                          std::vector<Idx2D> const& sequence_idx) {
+                          std::span<Idx2D const> sequence_idx) {
         if (!components.empty()) {
             update_component<CompType, CacheType>(components.begin(), components.end(), sequence_idx);
         }
     }
 
+    template <class CompType, cache_type_c CacheType>
+    void update_component(ConstDataset const& update_data, Idx pos, std::span<Idx2D const> sequence_idx) {
+        assert(construction_complete_);
+        assert(update_data.get_description().dataset->name == std::string_view("update"));
+
+        if (update_data.is_columnar(CompType::name)) {
+            this->update_component<CompType, CacheType>(
+                update_data.get_columnar_buffer_span<meta_data::update_getter_s, CompType>(pos), sequence_idx);
+        } else {
+            this->update_component<CompType, CacheType>(
+                update_data.get_buffer_span<meta_data::update_getter_s, CompType>(pos), sequence_idx);
+        }
+    }
+
     // update all components
     template <cache_type_c CacheType>
+    void
+    update_component(ConstDataset const& update_data, Idx pos,
+                     std::array<std::reference_wrapper<std::vector<Idx2D> const>, n_types> const& sequence_idx_map) {
+        run_functor_with_all_types_return_void([this, pos, &update_data, &sequence_idx_map]<typename CT>() {
```

Comment:
`ComponentType` has a special meaning because it is a variadic template argument `typename... ComponentType` to the `MainModelImpl` class. It therefore signifies (and requires) variadic expansion. `CompType` or `CT` can indeed be homogenized in most cases. Not sure we need to do that in this PR, though. We can discuss later today.

Reply:
changing this would introduce a significant amount of reformatting changes and I think it's better to hold that off

Reply:
This is for sure something, if we are doing it, that should go to a different PR. I see now that `ComponentType` is a variadic template argument, but it is not obvious until you reach the `functor_something` functions. So I would argue that we should make it more explicit and the homogenize the other two. Let's indeed discuss in later today.

Reply:
Let's make a separate warring thread for this topic.

Reply:
is this still needed after the discussion during refinement? if so, please take care of this